### ISSUE TITLE
💅 False positive on `useExportType`: exporting identifier with const + type

### ISSUE THREAD

**Issue by siketyan** [biomejs/biome#5382]
### Environment information

```bash
CLI:
  Version:                      
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.6.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /Users/[REDACTED]/biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noBlankTarget
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidLang
    complexity/noBannedTypes
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noMultipleSpacesInRegex
    complexity/noUselessCatch
    complexity/noUselessEmptyExport
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noWith
    complexity/useArrowFunction
    complexity/useRegexLiterals
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noFlatMapIdentity
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedLabels
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useIsNan
    correctness/useValidForDirection
    correctness/useYield
    nursery/noCommonJs
    nursery/noIrregularWhitespace
    nursery/noRestrictedImports
    nursery/noUselessEscapeInRegex
    nursery/useAdjacentOverloadSignatures
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noArguments
    style/noCommaOperator
    style/noInferrableTypes
    style/noNamespace
    style/noNonNullAssertion
    style/useAsConstAssertion
    style/useConst
    style/useEnumInitializers
    style/useExponentiationOperator
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodejsImportProtocol
    style/useNumberNamespace
    style/useNumericLiterals
    style/useSelfClosingElements
    style/useShorthandFunctionType
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAsyncPromiseExecutor
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noPrototypeBuiltins
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noThenProperty
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useNamespaceKeyword
    suspicious/useValidTypeof

Workspace:
  Open Documents:               0
```

### Rule name

useExportType

### Playground link

https://biomejs.dev/playground/?lintRules=all&analyzerFixMode=safeAndUnsafeFixes&code=YwBvAG4AcwB0ACAARQByAHIAbwByAEMAbwBkAGUAIAA9ACAAewAKACAAIABpAG4AdgBhAGwAaQBkAEEAcgBnAHUAbQBlAG4AdAA6ACAAIgBpAG4AdgBhAGwAaQBkAC0AYQByAGcAdQBtAGUAbgB0ACIALAAKACAAIABpAG4AdABlAHIAbgBhAGwARQByAHIAbwByADoAIAAiAGkAbgB0AGUAcgBuAGEAbAAtAGUAcgByAG8AcgAiACwACgB9ADsACgAKAHQAeQBwAGUAIABFAHIAcgBvAHIAQwBvAGQAZQAgAD0AIAAoAHQAeQBwAGUAbwBmACAARQByAHIAbwByAEMAbwBkAGUAKQBbAGsAZQB5AG8AZgAgAHQAeQBwAGUAbwBmACAARQByAHIAbwByAEMAbwBkAGUAXQA7AAoACgBlAHgAcABvAHIAdAAgAHsARQByAHIAbwByAEMAbwBkAGUAfQA7AAoA

### Expected result

No error.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
Workaround:
https://biomejs.dev/playground/?lintRules=all&analyzerFixMode=safeAndUnsafeFixes&code=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABFAHIAcgBvAHIAQwBvAGQAZQAgAD0AIAB7AAoAIAAgAGkAbgB2AGEAbABpAGQAQQByAGcAdQBtAGUAbgB0ADoAIAAiAGkAbgB2AGEAbABpAGQALQBhAHIAZwB1AG0AZQBuAHQAIgAsAAoAIAAgAGkAbgB0AGUAcgBuAGEAbABFAHIAcgBvAHIAOgAgACIAaQBuAHQAZQByAG4AYQBsAC0AZQByAHIAbwByACIALAAKAH0AOwAKAAoAZQB4AHAAbwByAHQAIAB0AHkAcABlACAARQByAHIAbwByAEMAbwBkAGUAIAA9ACAAKAB0AHkAcABlAG8AZgAgAEUAcgByAG8AcgBDAG8AZABlACkAWwBrAGUAeQBvAGYAIAB0AHkAcABlAG8AZgAgAEUAcgByAG8AcgBDAG8AZABlAF0AOwAKAAoA

**Comment by siketyan** [comment]
Note: This is happened in Biome preview build @ 5d73b1d. No error was reported in Biome v1.9.4.
